asistente_ladm_col/utils/quality.py

```diff
@@ -60,21 +74,76 @@
                                            BFS_TABLE_BOUNDARY_POINT_FIELD,
                                            MOREBFS_TABLE_BOUNDARY_FIELD,
                                            MORE_BOUNDARY_FACE_STRING_TABLE,
+                                           MUNICIPALITY_FIELD,
                                            LESS_TABLE,
                                            PLOT_TABLE,
                                            RIGHT_OF_WAY_TABLE,
-                                           SURVEY_POINT_TABLE)
+                                           SURVEY_POINT_TABLE,
+                                           ZONE_FIELD)
 
 class QualityUtils(QObject):
+    log_quality_message_emitted = pyqtSignal(str, int)
+    log_quality_button_emitted = pyqtSignal()
+    log_quality_message_ini_emitted = pyqtSignal(str)
+    log_quality_message_finish_emitted = pyqtSignal(str)
 
     def __init__(self, qgis_utils):
         QObject.__init__(self)
         self.qgis_utils = qgis_utils
         self.logic = LogicChecks()
         self.project_generator_utils = ProjectGeneratorUtils()
         self.log = QgsApplication.messageLog()
+        self.log_dialog_quality_text_content = ""
+        self.total_time = 0
+
+    def count_topology_rules(self, count):
+        self.log_quality_message_emitted.emit(QCoreApplication.translate("QualityUtils",""), count)
+        QCoreApplication.processEvents()
```

Comment:
`processEvents()` either here or in `asistente_ladm_col_plugin.py`, but not in both.